### PR TITLE
Limit objects accepted to be only log4j objects

### PR DESCRIPTION
**Pull request by jordansissel**

### CODE REVIEW

File: lib/logstash/inputs/log4j.rb [hunk 1]
```diff
@@ -197,4 +195,25 @@ def build_client_socket
...
+    def safety_check(name)
+      raise java.io.InvalidObjectException.new("Object type #{name} is not allowed.") if !ALLOWED_CLASSES.include?(name)
+    end
+
+    def resolveClass(desc)
+      name = desc.getName
+      safety_check(name)
+      return java.lang.Class.forName(name, false, @class_loader)
+    rescue java.lang.ClassNotFoundException => e
+      return super
```
Comment (suyograo) [inline lib/logstash/inputs/log4j.rb]: Why `return super` here? Shouldn't we just propagate `ClassNotFoundException` to the caller?

File: spec/inputs/log4j_spec.rb [hunk 2]
```diff
@@ -177,4 +180,21 @@
...
+    let(:data) {
+      oos.writeObject(input)
+      baos.toByteArray()
+    }
+
+    let(:bais) { java.io.ByteArrayInputStream.new(data) }
+    let(:ois) { LogStash::Inputs::Log4j::Log4jInputStream.new(bais) }
+
+    it "should reject non-log4j objects" do
+      expect { ois.readObject }.to raise_error(java.io.InvalidObjectException)
```
Comment (suyograo) [inline spec/inputs/log4j_spec.rb]: nice!

File: lib/logstash/inputs/log4j.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@
 require "socket"
 require "timeout"
 require 'logstash-input-log4j_jars'
+#require "logstash/inputs/log4j/log4j_input_stream"
```
Comment (suyograo) [inline lib/logstash/inputs/log4j.rb]: remove this commented require if we are not using it?

File: lib/logstash/inputs/log4j.rb (hunk 1 above)
Comment (jordansissel) [inline lib/logstash/inputs/log4j.rb]: Hmm, true. I don't remember why I added this. I'll remove it.

**Comment by jordansissel** [comment]
Revised; also pushed as a single commit.

**Pull request closed without merging**